.ci/test_r_package_valgrind.sh

```diff
@@ -14,6 +14,7 @@ RDvalgrind \
 
 cat ${ALL_LOGS_FILE}
 
+echo "writing valgrind output to ${VALGRIND_LOGS_FILE}"
```

Comment:
/gha run r-valgrind

Reply:
https://github.com/microsoft/LightGBM/actions/runs/456813525

Reply:
/gha run r-valgrind

Reply:
https://github.com/microsoft/LightGBM/runs/1635488178?check_suite_focus=true

Reply:
ok I think the fix in this PR fixed it! Now this output at the end of the logs shows the correct numbers, so I can tell the stderr redirection is working

```text
writing valgrind output to valgrind-logs.log
valgrind found 0 bytes definitely lost
valgrind found 0 bytes indirectly lost
valgrind found 336 bytes possibly lost
```